hack/terraform/aws/delete_dependency.sh

```diff
@@ -0,0 +1,57 @@
+#! /bin/bash
+# Copyright 2022 Antrea Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Delete security groups and the dependencies on AWS created by users.
+vpc_id="$( terraform output vpc_id | sed 's/\"//g' )"
+region="$( terraform output region | sed 's/\"//g' )"
+
+sgs="$( aws ec2 describe-security-groups --region $region --filters Name=vpc-id,Values=$vpc_id | jq )"
+length="$( echo $sgs | jq '.SecurityGroups|length' )"
+end=`expr $length - 1`
+
+function delete_SG(){
+    instances="$( aws ec2 describe-instances --region $region --filters Name=vpc-id,Values=$vpc_id Name=instance.group-id,Values=$1  | jq ".[][].Instances[].InstanceId" )"
+    
+    for instance in $instances; do
+        instance_ID="$( echo $instance | sed 's/\"//g' )"
+        aws ec2 terminate-instances --instance-ids $instance_ID
+        if [ $? -ne 0 ]; then
+            echo Failed to teminate instance $instance_ID
+        fi
+        sleep 60
+    done
+    
+    aws ec2 delete-security-group --group-id $1
+    if [ $? -ne 0 ]; then
+        echo Failed to delete Security Group $1
+    fi
+}
+
+
+for i in $(seq 0 $end); do
+    sg="$( echo $sgs | jq -r ".SecurityGroups[$i]" )"
+    sg_Tags="$( echo $sgs | jq -r ".SecurityGroups[$i].Tags?" )"
+    
+    if [[ "$sg_Tags" == "null" ]]; then
+        sg_ID="$( echo $sg | jq ".GroupId" | sed 's/\"//g' )"
+        delete_SG $sg_ID
+    else
+        sg_TF="$( echo $sg_Tags | jq -r '.[] | select(.Key=="Terraform", .Value=="true") ' )"
```

Comment:
This select actually perform OR and return the terraform tag twice. Should be
`echo $sg_Tags | jq -r '.[] | select(.Key=="Terraform" and .Value=="true")'`